core/app/services/uffizzi_core/deployment_service.rb

```diff
@@ -29,6 +29,25 @@ def create_from_compose(compose_file, project, user)
       deployment_form
     end
 
+    def update_from_compose(compose_file, project, user, deployment_id)
+      deployment_attributes = ActionController::Parameters.new(compose_file.template.payload)
+
+      deployment_form = UffizziCore::Api::Cli::V1::Deployment::UpdateForm.new(deployment_attributes)
+      deployment_form.assign_dependences!(project, user)
+      deployment_form.compose_file = compose_file
+
+      if deployment_form.valid?
+        deployment = UffizziCore::Deployment.find(deployment_id)
+        deployment.containers.destroy_all
+        deployment.compose_file.destroy if deployment.compose_file.kind.temporary?
+        deployment.update!(containers: deployment_form.containers, compose_file_id: compose_file.id)
```

Comment:
You have already invoked validation before this operation. If it does raise an error we need to wrap this logic to a transaction

Reply:
resolved in this MR https://github.com/UffizziCloud/uffizzi_app/pull/124